### PR TITLE
support parsing substring with ' ' space

### PR DESCRIPTION
**Pull request by elahehrashedi**
Add support parsing define variables that are in the form of a string, e.g., and also variables that include a space in the string:

```
target_compile_definitions(testPRa PUBLIC AWESOME="'pr a'")
target_compile_definitions(testPRb PUBLIC AWESOME="'pr b'")
set(CMAKE_CXX_COMPILER "c:/program files (x86)/mingw-w64/i686-8.1.0-posix-dwarf-rt_v6-rev0/mingw32/bin/g++.exe")

```
bug:
main:
https://github.com/microsoft/vscode-cmake-tools/issues/969
https://github.com/microsoft/vscode-cmake-tools/issues/1414

duplicates:
https://github.com/microsoft/vscode-cmake-tools/issues/1493
https://github.com/microsoft/vscode-cmake-tools/issues/1565

### CODE REVIEW

File: src/shlex.ts [hunk 1]
```diff
@@ -41,26 +42,38 @@ export function* split(str: string, opt?: ShlexOptions): Iterable<string> {
...
       }
-
       // Another quoted char
       token = (token || '') + char;
       continue;
     }
 
     if (quoteChars.includes(char)) {
+      // Beginning of a sub-quoted token
+      if (i > 0 && (opt.mode === 'posix' ? (str.substring(i - 1, i + 1) === "\\\'" || str.substring(i - 1, i + 1) === "\\\"") : str.substring(i - 1, i + 1) === "\\\"")) {
```
Comment (michelleangela) [inline src/shlex.ts:61]: `str.substring(i - 1, i + 1)` and `"\\\""` are repeated. Can this be simplified to:

```
const quote: string = str.substring(i - 1, i + 1);
if (i > 0 && (quote === "\\\"" || ( opt.mode === 'posix' && quote === "\\\'")))

**Review by michelleangela — CHANGES_REQUESTED** [review]

File: src/shlex.ts (hunk 1 above)
Comment (elahehrashedi) [reply src/shlex.ts]: I ended up removing this part

File: src/shlex.ts [hunk 2]
```diff
@@ -30,35 +28,33 @@ export function* split(str: string, opt?: ShlexOptions): Iterable<string> {
...
     }
 
-    if (quoteChars.includes(char)) {
-      // Beginning a new quoted token
-      quoteChar = char;
-      token = '';
+    if (i > 0 && quoteChars.includes(char)) {
+      // Beginning of a sub-quoted token
+      isSubQuote = true;
+      //quoteChar = char;
```
Comment (michelleangela) [inline src/shlex.ts:51]: commented code `//quoteChar = char;` at line 51. Remove?

**Review by michelleangela — CHANGES_REQUESTED** [review]

**Comment by elahehrashedi** [comment]
The test from the wrong input file was running. I had to recreate the branch.

**Pull request closed without merging**